src/components/Transactions/transactions.js

```diff
@@ -16,120 +16,202 @@ import { getTransactionsQuery } from '../../services/Transaction/transaction';
 import { Badge } from '@mui/material';
 import { TransactionType } from '../../utils/utils/constants';
 import { UseGetEntity } from '../../services/Entity/entity';
+import DateTimePicker from '@mui/lab/DateTimePicker';
 
 const Transactions = () => {
-    const auth = useAuth();
-    const { accountId } = useParams();
-    const [accountState, setAccountsState] = useState(undefined);
-    const [isLoading, setIsLoading] = useState(false);
-    const [open, setOpen] = useState(false);
-    const handleOpen = () => setOpen(true)
-    const handleClose = () => setOpen(false);
-    let history = useHistory();
-    let transactionsState = UseGetEntity(getTransactionsQuery(accountId));
+  const auth = useAuth();
+  const { accountId } = useParams();
+  const [accountState, setAccountsState] = useState(undefined);
+  const [isLoading, setIsLoading] = useState(false);
+  const [open, setOpen] = useState(false);
+  const handleOpen = () => setOpen(true)
+  const handleClose = () => setOpen(false);
+  let history = useHistory();
+  const initialFiler = {
+    type: 3,
+    date: {
+      from: undefined,
+      to: undefined
+    }
+  };
+  const [filter, setFilter] = useState(initialFiler);
+  let transactionsQuery = getTransactionsQuery(accountId, filter);
+  let transactionsState = UseGetEntity(transactionsQuery, filter);
+  const transactionTypeOptions = [
+    {
+      value: 1,
+      label: 'Depósito',
+    },
+    {
+      value: 2,
+      label: 'Retiro',
+    },
+    {
+      value: 3,
+      label: 'Todas',
+    }
+  ];
 
-    useEffect(() => {
-        setIsLoading(true);
-        if(auth.data && !auth.isLoading) {
-            getAccountQuery(accountId).then(result =>
-              setAccountsState(result.data())
-            ).finally(() => setIsLoading(false));
-        }
-    }, [accountId, accountState]);
+  useEffect(() => {
+    setIsLoading(true);
+    if (auth.data && !auth.isLoading) {
+      getAccountQuery(accountId).then(result =>
+        setAccountsState(result.data())
+      ).catch(() => setAccountsState(undefined)
+      ).finally(() => setIsLoading(false)
+      )
+    }
+  }, [accountId]);
 
-    return (accountState ? <>
-        <Navbar />
-        <Box>
-            <Button className='back-button' onClick={() => history.push('/accounts')}><ArrowBackIosIcon/>Volver</Button>
-        </Box>
-        <div className="transactions">
-          <Card>
-            <CardContent sx={{ flex: '1 0 auto' }}>
+  return (accountState ? <>
+    <Navbar />
+    <Box>
+      <Button className='back-button' onClick={() => history.push('/accounts')}><ArrowBackIosIcon />Volver</Button>
```

Comment:
Usign button to change to another view it's ok, but just as a tip, in terms of html good practices, to navigate to another url, its better to use a link